newIDE/app/src/Profile/EditProfileDialog.js

```diff
@@ -57,6 +72,115 @@ export const getUsernameErrorText = (error: ?AuthError) => {
   return undefined;
 };
 
+const GitHubUsernameField = ({
+  badges,
+  achievements,
+  githubUsername,
+  onSetGithubUsername,
+  onUpdateGitHubStar,
+  disabled,
+}: {
+  badges: ?Array<Badge>,
+  achievements: ?Array<Achievement>,
+  githubUsername: string,
+  onSetGithubUsername: (username: string) => void,
+  onUpdateGitHubStar: (
+    githubUsername: string
+  ) => Promise<UpdateGitHubStarResponse>,
+  disabled: boolean,
+}) => {
+  const { showAlert } = useAlertDialog();
+
+  const hasGithubStarBadge =
+    !!badges && badges.some(badge => badge.achievementId === 'github-star');
+  const githubStarAchievement =
+    (achievements &&
+      achievements.find(achievement => achievement.id === 'github-star')) ||
+    null;
+
+  return (
+    <I18n>
+      {({ i18n }) => (
+        <TextFieldWithButtonLayout
+          renderButton={style => (
+            <RaisedButton
+              onClick={async () => {
```

Comment:
Could you put this long method in a callback in the component?

Reply:
Done